[Specs] Introduce `COMPILATION_CACHE_ENABLE_CACHING` for controlling all the compiler-specific caching settings

akyrtzi opened the pull request:
Also make the new setting, along with `COMPILATION_CACHE_ENABLE_DIAGNOSTIC_REMARKS` show up in settings UI.

akyrtzi commented:
@swift-ci Please test

benlangmuir commented on Sources/SWBUniversalPlatform/Specs/Clang LLVM 1.0.xcspec:
```diff
@@ -2970,6 +2970,16 @@
+            {
+                Name = "CLANG_ENABLE_COMPILE_CACHE";
+                Type = Enumeration;
+                Values = (
```
Does this constrain the values or only impact help text? I'm not clear what it means for NOT_SET

akyrtzi replied:
This is similar to `SWIFT_ENABLE_EXPLICIT_MODULES` [here](https://github.com/swiftlang/swift-build/blob/f67ead1006bedb2243a55b149184a65287899008/Sources/SWBUniversalPlatform/Specs/Swift.xcspec#L923).

Is there a concern @owenv ?

benlangmuir replied:
Does the fact we are setting a new default value for the compiler-specific settings mean that NOT_SET will never be used for these anymore? If so maybe they can be changed to Boolean?

owenv replied:
It's been a while since I last looked at this, but I think it's safe to change this to a boolean setting since it's always going to get $(COMPILATION_CACHE_ENABLE_CACHING) as a backstop now

akyrtzi replied:
I've changed `CLANG_ENABLE_COMPILE_CACHE` and `SWIFT_ENABLE_COMPILE_CACHE` to boolean macros.

akyrtzi commented:
@swift-ci Please test

akyrtzi commented:
Changes since last time:
* Made `CLANG_ENABLE_COMPILE_CACHE` and `SWIFT_ENABLE_COMPILE_CACHE` boolean macros.
* Newly introduced `COMPILATION_CACHE_ENABLE_CACHING` also gets a default value from `COMPILATION_CACHE_ENABLE_CACHING_DEFAULT`
